packages/graphql/lib/services/resolvers-explorer.service.ts

```diff
@@ -172,24 +176,15 @@ export class ResolversExplorerService extends BaseExplorerService {
           undefined,
           args,
         );
-        let contextId: ContextId;
-        if (gqlContext && gqlContext[REQUEST_CONTEXT_ID]) {
-          contextId = gqlContext[REQUEST_CONTEXT_ID];
-        } else if (
-          gqlContext &&
-          gqlContext.req &&
-          gqlContext.req[REQUEST_CONTEXT_ID]
-        ) {
-          contextId = gqlContext.req[REQUEST_CONTEXT_ID];
-        } else {
-          contextId = createContextId();
-          Object.defineProperty(gqlContext, REQUEST_CONTEXT_ID, {
-            value: contextId,
-            enumerable: false,
-            configurable: false,
-            writable: false,
-          });
-        }
+
+        const contextId: ContextId =
+          this.contextIdFactory.getByRequest(gqlContext);
```

Comment:
I realized that this part of the code looked like [these](https://github.dev/nestjs/nest/blob/1ae72e1fdb338a6e00ac0ca19615602a91edc5bd/packages/core/middleware/middleware-module.ts#L212) lines from `middleware-module` in the core package. In the end, the intention of this part of the code was apparently to exactly the contextId from the request (in this case, the graphql context)

Reply:
@thiagomini instead of refactoring the existing code to use the injected factory, could we simply use the `ContextIdFactory` static class from the `@nestjs/core`? As I've mentioned on the call, although it's feasible for the `@nestjs/graphql` package, we can't really rely on the container in this case as in microservices & router-explorer (REST), we construct routes outside of the container (so at the end of the day, even if we introduce "custom context id factory" feature, it will have to be applied **outside** of the DI container anyway).

Otherwise, fantastic job Thiago!

Reply:
Hmmm yea, we could use the static ContextIdFactory here, but the problem is the implementation of the `getByRequest()` from `ContextIdFactory` would not work here, since it was created with the RESTful context in mind. If we look at the [implementation](https://github.dev/nestjs/nest/blob/7ac42b08ef31350d51b87d618be333fc69c1414f/packages/core/helpers/context-id-factory.ts#L28), it is trying to retrieve the `CONTEXT_ID` directly from the `request` object or `request.raw`, while in the graphql context we are trying to retrieve from the `request` or `request.req`.

With the above in mind, we could try to import the ContextIdFactory and override its implementation for `getByRequest` method, I suppose? But I feel this would be a work-around, instead of a long-term solution 🤔 I have another suggestion, since we can't have this injected via the DI container in some cases: 

1. We maintain the factory as a private attribute for this class, also as an interface, to avoid dependency of an implementation.
2. We give it a default value in the constructor, for the case DI system can't find a provider for it: a new class of the GraphlQLContextIDFactory

My main concern here is to avoid depending on such concrete classes and leverage the customization of such ContextIdFactory. Do you think it's feasible?